backend/tests/controllers/test_executions/test_logic.py

```diff
@@ -0,0 +1,100 @@
+from sqlalchemy.orm import Session
+
+from test_observer.controllers.test_executions.logic import (
+    get_matching_artefact_build_ids,
+    get_matching_test_execution_ids,
+)
+from test_observer.data_access.models import (
+    Artefact,
+    ArtefactBuild,
+    Environment,
+    Stage,
+    TestExecution,
+)
+
+
+def test_get_matching_artefact_build_ids(db_session: Session):
+    stage = db_session.query(Stage).filter(Stage.name == "beta").first()
+
+    artefact_one = Artefact(name="some artefact", version="1.0.0", stage=stage)
+    artefact_build_one = ArtefactBuild(architecture="some arch", artefact=artefact_one)
+
+    db_session.add_all([artefact_one, artefact_build_one])
+    db_session.commit()
+
+    artefact_two = Artefact(name="some artefact", version="1.0.1", stage=stage)
+    artefact_build_two = ArtefactBuild(architecture="some arch", artefact=artefact_two)
+    artefact_build_three = ArtefactBuild(
+        architecture="another arch", artefact=artefact_two
+    )
+
+    db_session.add_all([artefact_two, artefact_build_two, artefact_build_three])
+    db_session.commit()
+
+    artefact_build_ids = get_matching_artefact_build_ids(
+        session=db_session,
+        artefact=artefact_one,
+        architecture=artefact_build_two.architecture,
+    )
+
+    assert artefact_build_ids == [artefact_build_two.id, artefact_build_one.id]
+
+
+def _get_test_execution(
+    db_session: Session, environment: Environment, ci_link: str
+) -> TestExecution:
```

Comment:
what do you think about placing this function before other functions so we don't mix tests and helpers?

Reply:
Good idea! I have now moved it on the top of the tests